beacon-chain/powchain/service.go

```diff
@@ -171,9 +175,26 @@ func (w *Web3Service) ChainStartDeposits() []*pb.Deposit {
 	return w.chainStartDeposits
 }
 
-// Status always returns nil.
-// TODO(1204): Add service health checks.
+// Status is service health checks. Return nil or error.
 func (w *Web3Service) Status() error {
+	// Web3Service don't start
+	if !w.isRunning {
+		return nil
+	}
+	// get error from run function
+	if w.runError != nil {
+		return w.runError
+	}
+	// use a 1 minute  timeout for syncing process (if start syncing from old block)
+	minuteTimeout := time.Now().Add(-time.Minute)
+	// use a 5 minutes timeout for block time, because the max mining time is 278 sec (block 7208027)
+	// (analyzed the time of the block from 2018-09-01 to 2019-02-13)
+	fiveMinutesTimeout := time.Now().Add(-5 * time.Minute)
+	// check that web3 client is syncing
+	if w.lastHeadInteractTime.Before(minuteTimeout) &&
```

Comment:
wouldn't it be more accurate to have ` if w.blockTime.Before(fiveMinutesTimeout) { ...` only ? Are there any situations where `w.blockTime.Before(fiveMinutesTimeout)` is true but `w.lastHeadInteractTime.Before(minuteTimeout)` is false ?

Reply:
On a second thought , it might be a rare case but it is possible to have `w.lastHeadInteractTime.Before(minuteTimeout)` as false and `w.blockTime.Before(fiveMinutesTimeout)` as true. 
I think `w.blockTime.Before(fiveMinutesTimeout)` by itself is a valid error status, you could return a error if this happens by itself

Reply:
Yes, if we run Geth and load the blocks from start time of the ethereum network. We will get the block with the old time, but if the synchronization goes and we get a block per minute at least then the service works fine. In this situation we get old block time and good last iteraction time.

Reply:
We will not be loading blocks from the start time of the ETH 1.0 chain, whenever we start subscribing we will always request for the current head of 1.0 chain

Reply:
I removed the lastHeadInteractTime